internal/data/payments.go

```diff
@@ -2,15 +2,107 @@ package data
 
 import (
 	"context"
+	"database/sql"
 	"fmt"
+	"time"
 
 	"github.com/stellar/wallet-backend/internal/db"
+
+	"github.com/jmoiron/sqlx"
 )
 
 type PaymentModel struct {
 	db db.ConnectionPool
 }
 
+type Payment struct {
+	OperationID     int64
+	OperationType   string
+	TransactionID   int64
+	TransactionHash string
+	From            string
+	To              string
+	SrcAssetCode    string
+	SrcAssetIssuer  string
+	SrcAmount       int64
+	DestAssetCode   string
+	DestAssetIssuer string
+	DestAmount      int64
+	CreatedAt       time.Time
+	Memo            string
+}
+
+func (m *PaymentModel) GetLatestLedgerSynced(ctx context.Context, cursorName string) (uint32, error) {
+	var lastSyncedLedger uint32
+	err := m.db.QueryRowxContext(ctx, `SELECT value FROM ingest_store WHERE key = $1`, cursorName).Scan(&lastSyncedLedger)
+	// First run, key does not exist yet
+	if err == sql.ErrNoRows {
+		return 0, nil
+	}
+	if err != nil {
+		return 0, err
+	}
+
+	return lastSyncedLedger, nil
+}
+
+func (m *PaymentModel) UpdateLatestLedgerSynced(ctx context.Context, cursorName string, ledger uint32) error {
+	const query = `
+		INSERT INTO ingest_store (key, value) VALUES ($1, $2)
+		ON CONFLICT (key) DO UPDATE SET value = excluded.value
+	`
+	_, err := m.db.ExecContext(ctx, query, cursorName, ledger)
+	if err != nil {
+		return fmt.Errorf("updating last synced ledger to %d: %w", ledger, err)
+	}
+
+	return nil
+}
+
+func (m *PaymentModel) BeginTx(ctx context.Context) (*sqlx.Tx, error) {
+	tx, err := m.db.BeginTxx(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("beginning transaction: %w", err)
+	}
+
+	return tx, nil
+}
```

Comment:
Why do we need this function?

Reply:
Please, read my comment about the database transaction. We don't need to do everything on this PR, but I believe that's a better approach when handling database transactions.

Reply:
We used because the service didn't have the DB directly injected into it, but I've now made the `DB` in the model exported. Also, removed this function and implemented the DB transaction wrapper function as you suggested in the other comment.